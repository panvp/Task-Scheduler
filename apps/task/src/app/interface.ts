export interface taskListType{
    title: string;
    desciption: string;
    dropdownValue: string;
}

export interface ViewAllTaskPropsType{
    allTaskList: Array<taskListType>;
    deleteTaskHandler: (index: number) => void;
    editTaskHandler: (index: number) => void;
    filterValue: string;
}
export interface newTaskPropsType{
    isEdit: boolean;
    setIsEdit: React.Dispatch<React.SetStateAction<boolean>>;
    indexToEdit: number;
    taskToBeEdited: taskListType; 
    setAllTaskList: React.Dispatch<React.SetStateAction<Array<taskListType>>>;
}

