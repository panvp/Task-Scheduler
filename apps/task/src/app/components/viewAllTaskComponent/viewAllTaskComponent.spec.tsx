import { render,screen } from '@testing-library/react';
import ViewAllTaskComponent from './viewAllTaskComponent.component';
import { taskListType } from '../../interface';

describe('Home Screen Test Cases', () => {
  beforeEach(() => {
    HTMLDialogElement.prototype.close = jest.fn();
    const filterValue = "All";
    const deleteTaskHandler = jest.fn();
    const editTaskHandler = jest.fn();
    const allTaskList: taskListType[] = [];
    render(
      <ViewAllTaskComponent
        filterValue={filterValue}
        deleteTaskHandler={deleteTaskHandler}
        editTaskHandler={editTaskHandler}
        allTaskList={allTaskList}
      />
    );
  });
  test('Snapshot testing', () => {
    const homeScreen = screen.getByTestId('viewAllComponent');
    expect(homeScreen).toMatchSnapshot();
  });
  
});
