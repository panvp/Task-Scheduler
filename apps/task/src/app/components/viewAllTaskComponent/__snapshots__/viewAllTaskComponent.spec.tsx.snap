// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Screen Test Cases Snapshot testing 1`] = `
<div
  class="flex flex-wrap mt-4 mb-4 justify-center"
  data-testid="viewAllComponent"
/>
`;
